app/models/event_discount.rb

```diff
@@ -1,12 +1,13 @@
 # License: AGPL-3.0-or-later WITH Web-Template-Output-Additional-Permission-3.0-or-later
 class EventDiscount < ApplicationRecord
-  attr_accessible \
-    :code,
-    :event_id,
-    :name,
-    :percent
 
-  belongs_to :event
+  belongs_to :event, required: true
   has_many :tickets
 
+  validates_presence_of :code, :name, :percent
+
+  validates :percent, numericality: {greater_than: 0, less_than_or_equal_to: 100}
```

Comment:
Moved the validations out of the controller and into the model, where they should be.